packages/@sanity/default-layout/src/defaultLayout/DefaultLayout.tsx

```diff
@@ -0,0 +1,157 @@
+import {RouteScope} from '@sanity/base/router'
+import absolutes from 'all:part:@sanity/base/absolutes'
+import userStore from 'part:@sanity/base/user'
+import {LegacyLayerProvider} from '@sanity/base/components'
+import {useCurrentUser} from '@sanity/base/hooks'
+import Sidecar from '../addons/Sidecar'
+import RenderTool from '../main/RenderTool'
+import {CreateDocumentDialog} from '../createDocumentDialog'
+import {SchemaErrorReporter} from '../schemaErrors/SchemaErrorReporter'
+import {SideMenu} from '../sideMenu'
+import getNewDocumentModalActions from '../util/getNewDocumentModalActions'
+import {Tool} from '../types'
+import {Navbar} from '../navbar'
+import {useDefaultLayoutRouter} from '../useDefaultLayoutRouter'
+import {RootFlex, MainAreaFlex, ToolBox, SidecarBox, PortalDiv} from './styles'
+import {LoadingScreen} from './LoadingScreen'
+
+interface Props {
+  tools: Tool[]
+}
+
+export const DefaultLayout = (props: Props) => {
+  const {tools} = props
+  const router = useDefaultLayoutRouter()
+  const [createMenuIsOpen, setCreateMenuIsOpen] = useState<boolean>(false)
+  const [menuIsOpen, setMenuIsOpen] = useState<boolean>(false)
+  const [showLoadingScreen, setShowLoadingScreen] = useState<boolean>(true)
+  const [loaded, setLoaded] = useState<boolean>(false)
+  const [searchIsOpen, setSearchIsOpen] = useState<boolean>(false)
+  const [loadingScreenElement, setLoadingScreenElement] = useState<HTMLDivElement | null>(null)
+  const [portalElement, setPortalElement] = useState<HTMLDivElement | null>(null)
+  const {value: currentUser} = useCurrentUser()
+
+  useEffect(() => {
+    if (!loaded) {
+      setLoaded(true)
+    }
+  }, [loaded])
+
+  const handleAnimationEnd = useCallback(() => {
+    setShowLoadingScreen(false)
+  }, [])
+
+  useEffect(() => {
+    if (loadingScreenElement) {
+      loadingScreenElement.addEventListener('animationend', handleAnimationEnd, false)
+    }
+    return () => {
+      if (loadingScreenElement) {
+        loadingScreenElement.removeEventListener('animationend', handleAnimationEnd, false)
+      }
+    }
+  }, [handleAnimationEnd, loadingScreenElement])
+
+  const handleCreateButtonClick = useCallback(() => {
+    setCreateMenuIsOpen((prev) => !prev)
+  }, [])
+
+  const handleActionModalClose = useCallback(() => {
+    setCreateMenuIsOpen(false)
+  }, [])
+
+  const handleToggleMenu = useCallback(() => {
+    setMenuIsOpen((prev) => !prev)
+  }, [])
+
+  const handleSwitchTool = useCallback(() => {
+    setMenuIsOpen(false)
+  }, [])
+
+  const handleLogout = useCallback(() => {
+    userStore.actions.logout()
+  }, [])
+
+  const handleSearchOpen = useCallback((open) => {
+    setSearchIsOpen(open)
+  }, [])
+
+  const renderContent = () => {
+    const tool = router.state?.tool || ''
+    const documentTypes = getNewDocumentModalActions().map((action) => action.schemaType)
+
+    return (
+      <RootFlex $isOverlayVisible={menuIsOpen} direction="column" data-test-id="DefaultLayout">
```

Comment:
The data property should be named `data-testid="DefaultLayout"` (just a convention).